make_datasets.py

```diff
@@ -14,8 +14,15 @@ def gen_regression(args):
                            bias=rs.normal(0, 3),
                            random_state=rs)
 
-    np.save(args.filex, X)
-    np.save(args.filey, y)
+    if args.test_samples != 0:
+        train_samples = args.samples - args.test_samples
```

Comment:
Some of this code is redundant. If `args.test_samples` is zero, `train_samples` would be the same as `args.samples`, so the train part of the train-test split would still be the exact same. It would be better to put this `train_samples = ...` calculation outside of the if-block, and also move the saving of `args.filex` and `args.filey` out.

Same goes for `gen_classification` and `gen_kmeans`.

Reply:
Code was corrected.